fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -143,44 +134,21 @@ void init() throws IOException {
 
         LOGGER.trace("Assembling a map of node primary types -> templates...");
         final ImmutableMap.Builder<String, Template> templatesMapBuilder = builder();
-        final Session session = sessionFactory.getInternalSession();
-        try {
-            // we search all of the possible node primary types and mixins
-            for (final NodeTypeIterator primaryNodeTypes =
-                         session.getWorkspace().getNodeTypeManager()
-                                 .getPrimaryNodeTypes(); primaryNodeTypes.hasNext();) {
-                final NodeType primaryNodeType =
-                    primaryNodeTypes.nextNodeType();
-                final String primaryNodeTypeName =
-                    primaryNodeType.getName();
-
-                // Create a list of the primary type and all its parents
-                final List<NodeType> nodeTypesList = new ArrayList<>();
-                nodeTypesList.add(primaryNodeType);
-                nodeTypesList.addAll(Arrays.asList(primaryNodeType.getSupertypes()));
-
-                // Find a template that matches the primary type or one of its parents
-                nodeTypesList.stream()
-                             .map(NodeType::getName)
-                             .filter(x -> !isBlank(x) && velocity.resourceExists(getTemplateLocation(x)))
-                             .findFirst()
-                             .ifPresent(x -> addTemplate(primaryNodeTypeName, x, templatesMapBuilder));
-            }
-
-            final List<String> otherTemplates =
-                    ImmutableList.of("jcr:nodetypes", "node", "fcr:versions", "fcr:fixity");
-
-            for (final String key : otherTemplates) {
-                final Template template =
-                    velocity.getTemplate(getTemplateLocation(key));
-                templatesMapBuilder.put(key, template);
-            }
-
-            templatesMap = templatesMapBuilder.build();
-
-        } catch (final RepositoryException e) {
-            throw new RepositoryRuntimeException(e);
-        }
+
+        of("jcr:nodetypes", "fcr:versions", "fcr:fixity", "binary", "resource", "root", "default")
+            .forEach(key -> templatesMapBuilder.put(key, velocity.getTemplate(getTemplateLocation(key))));
+
+        templatesMap = templatesMapBuilder.build();
+
+        final ImmutableMap.Builder<String, String> typeMapBuilder = builder();
+        typeMap = typeMapBuilder.put(REPOSITORY_NAMESPACE + "RepositoryRoot", "root")
```

Comment:
I agree with you there -- the issue I was trying to avoid was having the same `Template` object as the value of multiple `Map<String, Template>` entries. For example, fedora:Version, fedora:Pairtree and fedora:Container all need to result in the same `Template`. If that's not a concern, then I can certainly remove that `typeMap`.